src/main/java/net/fabricmc/loader/impl/ModContainerImpl.java

```diff
@@ -85,6 +103,25 @@ private Path obtainRootPath() {
 		}
 	}
 
+	@Override
+	public Optional<ModContainer> getContainingMod() {
+		return parentModId != null ? FabricLoaderImpl.INSTANCE.getModContainer(parentModId) : Optional.empty();
+	}
+
+	@Override
+	public Collection<ModContainer> getContainedMods() {
+		if (childModIds.isEmpty()) return Collections.emptyList();
+
+		List<ModContainer> ret = new ArrayList<>(childModIds.size());
+
+		for (String id : childModIds) {
+			ModContainer mod = FabricLoaderImpl.INSTANCE.getModContainer(id).orElse(null);
+			if (mod != null) ret.add(mod);
```

Comment:
Could do this as
```java
FabricLoaderImpl.INSTANCE.getModContainer(id).ifPresent(ret::add);
```

Reply:
that lambda captures though, I don't like these optional methods :D